kml/serdes_gpx.cpp

```diff
@@ -358,6 +365,127 @@ std::string GpxParser::BuildDescription() const
   return m_description + "\n\n" + m_comment;
 }
 
+namespace
+{
+
+std::optional<std::string> GetDefaultLanguage(LocalizableString const & lstr)
+{
+  auto const firstLang = lstr.begin();
+  if (firstLang != lstr.end())
+    return {firstLang->second};
+  return {};
+}
+
+std::string CoordToString(double c)
+{
+  std::ostringstream ss;
+  ss.precision(8);
+  ss << c;
+  return ss.str();
+}
+
+void SaveColorToRGB(Writer & writer, uint32_t rgba)
+{
+  writer << NumToHex(static_cast<uint8_t>(rgba >> 24 & 0xFF))
+         << NumToHex(static_cast<uint8_t>((rgba >> 16) & 0xFF))
+         << NumToHex(static_cast<uint8_t>((rgba >> 8) & 0xFF));
+}
+
+
+void SaveCategoryData(Writer & writer, CategoryData const & categoryData)
+{
+  writer << "<metadata>\n";
+  auto name = GetDefaultLanguage(categoryData.m_name);
+  if (name.has_value())
+    writer << kIndent2 << "<name>" << name.value() << "</name>\n";
+  auto description = GetDefaultLanguage(categoryData.m_description);
+  if (description.has_value())
```

Comment:
ditto

Reply:
In general, you don't need to call .has_value() for std::optional objects. if (obj) is enough.